src/GBA.c

```diff
@@ -529,7 +529,7 @@ static inline int dataTicksAccess16(u32 address) // DATA 8/16bits NON SEQ
     int waitState = value;
     if (!waitState)
       waitState = 1;
-    busPrefetchCount = ((++busPrefetchCount)<<waitState) - 1;
+    busPrefetchCount = ((busPrefetchCount + busPrefetchCount)<<waitState) - 1;
```

Comment:
this `fix` is not correct
supposedly it should be ```busPrefetchCount = ((busPrefetchCount+1)<<waitState) - 1;```
if the intended effect was achieved can be controlled by compiling the "old" version with -O0 and the new version with -O0 and compare if the generated object files are identical (for example using md5sum)